pyk/src/pyk/kcfg.py

```diff
@@ -351,21 +395,25 @@ def _short_label(label):
 
         return graph.source
 
-    def _resolve_all(self, short_id: str) -> List[str]:
-        return [node_id for node_id in self._nodes if compare_short_hashes(short_id, node_id)]
+    def _resolve_all(self, id_like: str) -> List[str]:
```

Comment:
I'd call this `_resolve_hash`, and keep its old behavior.

Then `_resolve_or_none` can be implemented as follows.
* If `id_like` is a hash, then delegate to `_resolve_hash`, and raise an exception if there are multiple results.
* Otherwise, we have an alias. If it starts with `@`, drop the first character.
* Try to resolve the alias.

As a result, `@` is optional, and can be used to enforce strings to be resolved as aliases (as something starting with `@` is never a hash). So for example
* `abcd` is resolved as a hash
* `@abcd` is resolved as alias `abcd`
* `hello` is resolved as alias `hello`
* `@hello` is resolved as alias `hello`

Optionally, we can also refine `_resolve` to give a more specific error message if resolution fails:
* If `id_like` is a hash: `Unknown node hash: ...`
* Otherwise: `Unknown alias: ...`

What do you think?

Reply:
I like the refactoring. But wasn't the whole point of adding the `@` to avoid confusion between the two.

If we feel the need for shorted aliases, I'm fine with adding that in a later PR, but this PR is blocking two other PRs I have for the ERC20 repo. Lets get the basics working and then iterate from there.

Reply:
> But wasn't the whole point of adding the @ to avoid confusion between the two?

My point is that if `id_like` is not a valid hash, we can omit the `@`, because we know ww have an alias. But this is just a small usability improvement.

In any case, the input should be validated to be either a hash or an alias (if `@` is optional, the latter does not have to be checked).